src/indexer/operation.rs

```diff
@@ -1,18 +1,24 @@
+use crate::query::Weight;
 use crate::schema::{Document, Term};
 use crate::Opstamp;
 
 /// Timestamped Delete operation.
-#[derive(Clone, Eq, PartialEq, Debug)]
 pub struct DeleteOperation {
     pub opstamp: Opstamp,
-    pub term: Term,
+    pub target: DeleteTarget,
+}
+
+/// Target of a Delete operation
+pub enum DeleteTarget {
+    Term(Term),
```

Comment:
Shouldn't we just use `Box<dyn Weight>` for `TermWeight` too?

Reply:
Hum, it might be trickier than I thought. [`test_deletequeue`](https://github.com/quickwit-oss/tantivy/pull/1535/files#diff-8ebb1eab4107a255705c22aac2a218ffbcaf3c7c6223d27ac9a9063eaebe2b8eR260) doesn't have access to a `Searcher`. I'm not sure how to make that test compile without creating a dummy index in ram.